src/lib/sessions.ts

```diff
@@ -44,16 +46,33 @@ export const saveSession = (session: Session): void => {
 	const currentSessions = get(sessionsStore) || [];
 
 	// Find the index of the session with the same id, if it exists
-	const existingSessionIndex = currentSessions.findIndex(s => s.id === session.id);
+	const existingIndex = currentSessions.findIndex((k) => k.id === session.id);
 
-	if (existingSessionIndex !== -1) {
+	if (existingIndex !== -1) {
 		// Update the existing session
-		currentSessions[existingSessionIndex] = session;
+		currentSessions[existingIndex] = session;
 	} else {
 		// Add the new session if it doesn't exist
 		currentSessions.push(session);
 	}
 
-	// Update the store, which will trigger the localStorage update
-	sessionsStore.set(currentSessions);
+	// Sort the sessions by updatedAt in descending order (most recent first)
+	const sortedSessions = sortStore(currentSessions);
+
+	// Update the store with the sorted sessions
+	sessionsStore.set(sortedSessions);
 };
+
+export function getSessionMetadata(session: Session) {
+	const metadata: { model: string; updatedAt?: string; knowledge?: string } = {
+		model: session.model
+	};
+	if (session.updatedAt) metadata.updatedAt = formatTimestampToNow(session.updatedAt);
+	if (session.knowledge?.name) metadata.knowledge = session.knowledge?.name;
+	return metadata;
+}
+
+export function formatSessionMetadata(session: Session) {
+	const subtitles: string[] = Object.values(getSessionMetadata(session));
+	return subtitles.join(' • ');
+}
```

Comment:
In cases like this I'd like to avoid having single-use abstractions.
If we are not calling `getSessionMetadata()` anywhere else move it's logic inside `formatSessionMetadata()`.

Reply:
You're right. I had done that for a previous version of the files and then I forgot to revert it to a single function.